bus-mapping/src/circuit_input_builder/input_state_ref.rs

```diff
@@ -1120,7 +1120,14 @@ impl<'a> CircuitInputStateRef<'a> {
                     false,
                     op,
                 );
-                self.tx.steps_mut()[step_index]
+                let idx = if from_precompile_call {
+                    // prev step is callop or begin_tx.
```

Comment:
are you sure the 'prev step is callop'?   inside callop, `state.handle_return(&mut precompile_step` is called first, then `Ok(vec![exec_step, precompile_step])` is returned and appended after ` state.tx.steps_mut()`.  So i am afraid the ` step_index - 1` here is the step **before** callop?